src/main/java/com/mongodb/hibernate/jdbc/MongoConnectionProvider.java

```diff
@@ -48,38 +48,34 @@
 import org.jspecify.annotations.Nullable;
 
 /**
- * {@linkplain com.mongodb.hibernate.dialect.MongoDialect MongoDB dialect}'s customized JDBC {@link ConnectionProvider}
- * SPI implementation.
+ * {@linkplain MongoDialect MongoDB dialect}'s customized JDBC {@link ConnectionProvider} SPI implementation.
  *
- * <p>{@link MongoConnectionProvider} uses the {@value JdbcSettings#JAKARTA_JDBC_URL} configuration property to create a
- * MongoDB <a href="https://www.mongodb.com/docs/manual/reference/connection-string/">connection string</a>. The
- * configuration property is mandatory if {@link MongoClientCustomizer} service is not used. Otherwise, a
- * {@link ConnectionString} instance will be provided to the
- * {@link MongoClientCustomizer#customize(MongoClientSettings.Builder, ConnectionString)} method as a reference
- * ({@code null} if {@value JdbcSettings#JAKARTA_JDBC_URL} is not configured), and it is up to the
- * {@code MongoClientCustomizer} to use it or not.
+ * <p>All the work done via a {@link Connection} {@linkplain MongoConnectionProvider#getConnection() obtained} from this
+ * {@linkplain ConnectionProvider} is done within the same {@link ClientSession}.
+ * {@linkplain ClientSession#startTransaction() MongoDB transactions} are used only if
+ * {@linkplain Connection#getAutoCommit() auto-commit} is disabled.
  *
- * @see ConnectionProvider
- * @see JdbcSettings#JAKARTA_JDBC_URL
- * @see MongoClientCustomizer
- * @see <a href="https://www.mongodb.com/docs/manual/reference/connection-string/">connection string</a>
+ * @see MongoDialectSettings
  */
 public final class MongoConnectionProvider
         implements ConnectionProvider, Configurable, Stoppable, ServiceRegistryAwareService {
-
     @Serial
     private static final long serialVersionUID = 1L;
 
+    private @Nullable MongoDialectSettings config;
     private @Nullable MongoClient mongoClient;
 
-    private @Nullable MongoClientCustomizer mongoClientCustomizer;
+    private @Nullable MongoDialectConfigurator mongoDialectConfigurator;
 
     @Override
     public Connection getConnection() throws SQLException {
         try {
             var client = assertNotNull(mongoClient);
             var clientSession = client.startSession();
-            return new MongoConnection(client, clientSession);
+            var initializedConfig = assertNotNull(config);
+            var connection = new MongoConnection(initializedConfig, client, clientSession);
+            connection.setAutoCommit(initializedConfig.isAutoCommit());
```

Comment:
the above step is confusing for Hibernate internally will set autoCommit based on the common configuration already. Should we duplicate what Hibernate has done (which might be more complicated than the above)?

You can test by removing the above statement and you can see Hibernate is working just as expected.

Reply:
It is up to a connection provider to respect `AvailableSettings.AUTOCOMMIT` or not. We discussed and agreed not to respect it.

Done in 0155ba09610e16853c3e6c9fbf41302186c39b8d.